drive/src/drive/object_size_info.rs

```diff
@@ -508,15 +522,31 @@ pub enum KeyValueInfo<'a> {
     /// A key by reference
     KeyRefRequest(&'a [u8]),
     /// Max size possible for value
-    KeyValueMaxSize((usize, usize)),
+    KeyValueMaxSize((u16, u32)),
 }
 
 impl<'a> KeyValueInfo<'a> {
+    /// Returns key ref request
+    pub fn as_key_ref_request(&'a self) -> Result<&'a [u8], Error> {
+        match self {
+            KeyRefRequest(key) => Ok(key),
+            KeyValueMaxSize((_, _)) => Err(Error::Drive(DriveError::CorruptedCodeExecution(
+                "requesting KeyValueInfo as key ref request however it is a key value max size",
+            ))),
+        }
+    }
+
     /// Returns key length
-    pub fn key_len(&'a self) -> usize {
+    pub fn key_len(&'a self) -> u16 {
         match self {
-            KeyRefRequest(key) => key.len(),
+            KeyRefRequest(key) => key.len() as u16,
             KeyValueMaxSize((key_size, _)) => *key_size,
         }
     }
 }
+
+/// Deletion Info
+pub struct DeletionInfo<'a, const N: usize> {
```

Comment:
Are we actually using it and its fields?

Reply:
No, I decided to keep it, since we will use it soon. If we don't we will remove it later.